jj/mock/_history/_history_representation.py

```diff
@@ -0,0 +1,57 @@
+import os
+import shutil
+from distutils.util import strtobool
+from pprint import pformat as pf
+from typing import List, Optional, Union
+
+from ._history_item import HistoryItem
+
+__all__ = ("default_history_repr", "HistoryType", "HistoryRepr",)
+
+HistoryType = Union[List[HistoryItem], None]
+
+REMOTE_MOCK_PPRINT = os.environ.get("JJ_REMOTE_MOCK_PPRINT", "True")
+REMOTE_MOCK_PPRINT_LIMIT = os.environ.get("JJ_REMOTE_MOCK_PPRINT_LIMIT", 1000000)
+REMOTE_MOCK_PPRINT_WIDTH = os.environ.get("JJ_REMOTE_MOCK_PPRINT_WIDTH")
+
+
+class HistoryRepr:
```

Comment:
If this class is designed for extension and used as the type `history_repr: Optional[HistoryRepr] = None`, it's a good idea to separate it into a base (abstract) class and a default (concrete) class:

```python
from abc import ABC, abstractmethod

class HistoryFormatter(ABC):
    @abstractmethod
    def format_history(self, history):
        pass

class PrettyHistoryFormatter(HistoryFormatter):
    # your implementation here
```

This way, you can easily extend the base class while maintaining a clear structure